pkg/ingester/errors.go

```diff
@@ -23,26 +23,18 @@ import (
 	"github.com/grafana/mimir/pkg/util/validation"
 )
 
+const (
+	integerUnavailableMsgFormat = "ingester is unavailable (current state: %s)"
+)
+
 var (
 	// This is the closest fitting Prometheus API error code for requests rejected due to limiting.
 	tooBusyError = newErrorWithHTTPStatus(
-		safeToWrapError("the ingester is currently too busy to process queries, try again later"),
+		errors.New("the ingester is currently too busy to process queries, try again later"),
 		http.StatusServiceUnavailable,
 	)
 )
```

Comment:
This error is returned called only from `checkReadOverloaded`, I would move it there (as the errors in this file don't return grpc nor http status codes) and modify the tests to make sure they assert on the status code of http-grpc, not on the error message.